CONTRIBUTING.md

```diff
@@ -1,12 +1,194 @@
+You can open an [issue](https://github.com/vertica/vertica-nodejs/issues)
+with details describing what feature(s) you would like added or changed.
+
+If you would like to implement the feature yourself, open an issue to ask before working on it. Once approved, please refer to the [Code Contributions](#code-contributions) section.
+
+# Code Contributions
+
+## Step 1: Fork
+
+Fork the project [on Github](https://github.com/vertica/vertica-nodejs) and check out your copy locally.
+
+```shell
+git clone git@github.com:YOURUSERNAME/vertica-nodejs.git
+cd vertica-nodejs
+```
+
+Your GitHub repository **YOURUSERNAME/vertica-nodjes** will be called "origin" in
+Git. You should also setup **vertica/vertica-nodejs** as an "upstream" remote.
+
+```shell
+git remote add upstream git@github.com:vertica/vertica-nodejs.git
+git fetch upstream
+```
+
+### Configure Git for the first time
+
+Make sure git knows your [name](https://help.github.com/articles/setting-your-username-in-git/ "Set commit username in Git") and [email address](https://help.github.com/articles/setting-your-commit-email-address-in-git/ "Set commit email address in Git"):
+
+```shell
+git config --global user.name "John Smith"
+git config --global user.email "email@example.com"
+```
+
+## Step 2: Branch
+
+Create a new branch for the work with a descriptive name:
+
+```shell
+git checkout -b my-fix-branch
+```
+
+## Step 3: Install dependencies and configure environment
+
+Note: If using the VS Code `Remote - Containers` extension. Skip this step and see the instructions in step 4
+
+If yarn is not already installed, install yarn: 
+
+```shell
+npm install -g yarn
+```
+
+Then we will let yarn install all the necessary dependencies for you.
+From your workspace root, run: 
+```shell
+yarn
+yarn lerna bootstrap
+```
+
+
+## Step 4: Get the test suite running
+
+Each package in the *vertica-nodejs* repository comes with a test suite of its own. It’s our policy to make sure all tests pass at all times. 
+
+Integration tests need to connect to a Vertica database to properly run, so you must have access to a Vertica database. If using VS Code, you can install the `Remote - Containers` extension and it will use the configuration under the `.devcontainer` folder to automatically create dev containers, including Vertica.  See [here](https://code.visualstudio.com/docs/remote/containers) for more information on developing in containers using VS Code.
+
+Spin up your Vertica database for integration tests. Once done you will need to provide the proper environment variables to run the existing test suite. See the following examples and update them to point to your running Vertica database: 
+
+  ```shell
+  # Set environment variables in linux
+  $ export V_HOST=10.0.0.2
+  $ export V_PORT=5433
+  $ export V_USER=dbadmin
+  $ export V_PASSWORD=xxxx
+  $ export V_DATABASE=VMart
+  $ export V_BACKUP_SERVER_NODE=10.0.0.3 (NOT REQUIRED)
```

Comment:
Should the default host be localhost and default password empty (if we assume most users/devs will be running Vertica CE in a Docker container when trying out the client)?

Reply:
I suppose so. These were just examples but might as well make them closer to what the contributor might be needing